utils/utils.h

```diff
@@ -163,13 +163,13 @@ extern struct gbConf gbConf;
             char timestamp[GB_TIME_STRING_BUFLEN] = {0};               \
             LOCK(gbConf.lock);                                         \
             if (level <= gbConf.logLevel) {                            \
-              if ((str) == "mgmt")                                     \
```

Comment:
Wow! Did I do this myself ? Can't believe :D

Reply:
> Wow! Did I do this myself ? Can't believe :D

Looks like this is from @amarts. Amar, is there any reason for comparing the const char strings this way ?

Reply:
Normally it works well, but if we pass the "str" without a const string, such as allocated from the heap when running, we may hit the bugs.

Reply:
`git show 9e38bb59` : for fixing the coverity issues. we were sending 'static' string as argument, and comparing it in strcmp() made the coverity tool crib about it.

Reply:
@amarts @lxbsz I recollect https://github.com/gluster/gluster-block/pull/171 was to fix some 50+ coverity issues.

@lxbsz may be better we makesure to pass the "str" with a const string is better or just cast the 'str'

if (((const char *)str) == "cli")

Reply:
@pkalever 
Still the same :

../utils/utils.h:166:39: warning: comparison with string literal results in unspecified behavior [-Waddress]
               if (((const char *)str) == "mgmt")                         \
                                       ^

Reply:
Hmm ... 
If coverity was complaining about using 'strcmp()' with using static strings.

And the rpmbuild complains about  'comparison with string literal  with =='

May be strdup the str and use it with strcmp ? does that make-sense to solve both the warnings from coverity and rpmbuild ?

@amarts @lxbsz ?

Reply:
@pkalever @amarts 
Let's make one inline help to do the check and then open the log files:

```
+static inline FILE *openLogFile(char *dom)
+{
+    FILE *fd = stderr;
+
+    if (!dom)
+        return fd;
+
+    if (!strcmp(dom, "mgmt")) {
+        fd = fopen (gbConf.daemonLogFile, "a");
+    } else if (!strcmp(dom, "cli")) {
+        fd = fopen (gbConf.cliLogFile, "a");
+    } else if (!strcmp(dom, "gfapi")) {
+        fd = fopen (gbConf.gfapiLogFile, "a");
+    } else if (strcmp(dom, "cmdlog")) {
+        fd = fopen (gbConf.cmdhistoryLogFile, "a");
+    }
+
+    return fd;
+}

```

Will this make sense ?

Reply:
> Hmm ...
> If coverity was complaining about using 'strcmp()' with static strings.
> 
> And the rpmbuild complains about 'comparison with string literal with =='
> 
> May be strdup the str and use it with strcmp ? does that make-sense to solve both the warnings from coverity and rpmbuild ?
> 
> @amarts @lxbsz ?

@pkalever @amarts 
Not sure whether the static inline will also have the same problem as the MACRO in coverity ?

Reply:
@lxbsz yeah, was thinking the same.

Reply:
@lxbsz did you thought about my proposal of strdup(str) above ? just mentioning here again, assuming you might have missed it.

Reply:
@pkalever 
Sorry.
I just thought this will be a bit ungraceful.
But I am okay with this if the static inline still won't work for us.
Thanks.

Reply:
@amarts Do you think static inline function using strcmp with static strings bypass the coverity checks ?

Else, I recommend to strdup the static strings and then compare.

Thanks!

Reply:
@pkalever 
Update it by using the GB_STRDUP, the static inline may also hit the same issue with some GCC optimal options.

Thanks.